### ISSUE TITLE
Reduce exception message for loading missing files?

### ISSUE THREAD

**Issue by rcomer** [SciTools/iris#2032]
Suppose I have

``` python
file_spec_a = 'path_to_files_a/*.nc'
file_spec_b = 'path_to_files_b/*.nc'
```

if `file_spec_b` doesn't point to any existing files, then `iris.load([file_spec_a, file_spec_b])` rightly produces an exception in [`iris.io.expand_filespecs`](https://github.com/SciTools/iris/blob/master/lib/iris/io/__init__.py#L165).  However, the exception message includes all the filenames matching `file_spec_a`, as well as telling me that `file_spec_b` expanded to empty.  This can sometimes be a lot of output, which I have to search through to find the "empty" so I can fix the problem.  Could this exception message be reduced to only list the file specifications which don't expand?

**Comment by ajdawson** [comment]
#2014 is actively addressing this code now, this would be useful input for that work. Ping @davidnmurray.

**Comment by pelson** [comment]
#2325 touched this most recently. I'm completely up for capping the length of the matched file list so that you aren't flooded when you have thousands of matches. I'd want to keep it there if you have a small number though.

**Comment by rcomer** [comment]
@pelson could you explain the rationale for including the matches?  My basic assumption is that the exception message only needs to tell the user about what went wrong, so including what went right seems superfluous.

**Comment by pelson** [comment]
That's a fair point. I guess the reason for showing all is that it will often be the case that if you're supplying multiple glob patterns they will probably follow a similar pattern, and it is useful to get feedback if one has worked but the other hasn't - normally this will occur in two scenarios:

 * the user has subtly changed their glob pattern, and it's broke/excludes more files than they intended
 * the glob pattern is fine, but the user has got the directory wrong

In either case, getting *some* feedback of a successful case is useful knowledge for fixing the broken case. I completely agree that the successes aren't quite as important as the fails though... so I'd be up for capping the length of the files matched...

**Comment by rcomer** [comment]
OK, that makes sense.  If it’s just to help the user understand the difference between what works and what doesn’t, would something like this suffice?
```
One or more of the files specified did not exist:
42 files match path_to_files_a/*.nc
path_to_files_b/*.nc expanded to empty
```

**Comment by rcomer** [comment]
Resolved by #2925.
